sdk/include/opentelemetry/sdk/metrics/aggregation/default_aggregation.h

```diff
@@ -159,7 +160,29 @@ class DefaultAggregation
               new DoubleSumAggregation(nostd::get<SumPointData>(point_data)));
         }
       default:
-        return DefaultAggregation::CreateAggregation(instrument_descriptor, nullptr);
+        return nullptr;  // won't reach here
```

Comment:
Add an assert here?

Reply:
Thanks, will look into this separately. Not a big fan of adding assert in telemetry sdk. But returning `nullptr` (even for unreachable code) is also not nice thing. Probaby we can return DropAggregation just to be cautious. Will create a issue to track this.